### ISSUE TITLE
CI tests not working in other repositories

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#251]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/251)
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Feb 13, 2018, 11:28

`.gitlab-ci.yml` currently sets `INTEGRATION_CACHE` to the absolute path */builds/BuildStream/buildstream/cache/*, which is the `cache` subdirectory of the working directory of the CI runner. However, the absolute path of the working directory depends on the URL of the repository, which means that CI tests no longer work in copies/forks of the main repository. We should fix `.gitlab-ci.yml` such that it doesn't rely on absolute paths.

Example of error message:
```
PermissionError: [Errno 13] Permission denied: '/builds/BuildStream'
```

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Feb 13, 2018, 17:50

mentioned in merge request !273

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Feb 13, 2018, 18:17

closed via merge request !273

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Feb 13, 2018, 18:17

closed via commit 955b184ea1a21bc3b0d81a1f6f7103b152157a58

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Feb 28, 2018, 15:14

mentioned in commit 955b184ea1a21bc3b0d81a1f6f7103b152157a58
